### PR TITLE
Update openapispec job to apply directly to the cluster

### PR DESCRIPTION
**Pull request by raakatz**
# Changes

### Removed ArgoCD hostname input
- set dynamically based on `stage` input
- needed only when the token secret is set

### Verify ArgoCD token input
- if going to deploy by ArgoCD (instead of committing to kube-manifests) or if going to update OpenAPI spec, verify that the token exists

### Move `get manifest info` job to init
- Gets variables needed by ArgoCD instead of reading from files. Super important for the multi-repo setup

### Apply OpenAPI spec directly to K8s
- Uses Teleport to authenticate
- applies directly to k8s, no need to push to git

# BREAKING!!!

Requires all repos that currently use `upload-open-api` to provide the `ARGOCD_TOKEN` secret.
I will open PRs for all the repos that use it, providing the secret and deleting the `service-profile.yaml` file from their directories https://github.com/monta-app/kube-manifests/pull/903

Progess:

- [x] ocpp https://github.com/monta-app/service-ocpp/pull/1142
- [x] api-v2-partner-api https://github.com/monta-app/partner-api/pull/1150
- [x] integrations https://github.com/monta-app/service-integrations/pull/689
- [x] data-broker
- [x] grid

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -125,6 +126,63 @@ jobs:
...
+        if: ${{ inputs.upload-open-api }}
+        shell: bash
+        env:
+          SERVICE_IDENTIFIER: ${{ inputs.service-identifier }}
+          STAGE: ${{ inputs.stage }}
+          ARGOCD_TOKEN: ${{ secrets.ARGOCD_TOKEN }}
+          ARGOCD_SERVER: ${{ steps.set-argocd-server.outputs.argocd-hostname }}
+        run: |
+          if [ "${{ secrets.ARGOCD_TOKEN }}" == '' ]; then
+            echo "MUST INPUT ARGOCD_TOKEN SECRET"
```
Comment (HotThoughts) [inline .github/workflows/deploy.yaml:160]:
```suggestion
            echo "::error::MUST INPUT ARGOCD_TOKEN SECRET"
```

All look great. If I have to say something, [workflow commands](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions) will be it. It is just a visual improvement over stdout, giving you a red error message.

**Review by HotThoughts — APPROVED** [review]

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (raakatz) [reply .github/workflows/deploy.yaml:160]: Oh that's amazing, didn't know about that!

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -597,10 +615,11 @@ jobs:
...
-          ./argocd --auth-token $ARGOCD_TOKEN --server ${{ inputs.argocd-server }} app set argocd/${{ inputs.service-identifier }}-${{ inputs.stage }} -p kotlin.revision=${GITHUB_SHA::8} -p kotlin.build=${{ github.run_number }} -p kotlin.image.tag=${GITHUB_SHA}
+          OLD_TAG=$(./argocd --grpc-web --auth-token $ARGOCD_TOKEN --server $ARGOCD_SERVER app get argocd/${{ inputs.service-identifier }}-${{ inputs.stage }} --show-params -o yaml | yq -e=0 '.spec.source.helm.parameters[] | select(.name == "image.tag").value')
+          ./argocd --grpc-web --auth-token $ARGOCD_TOKEN --server $ARGOCD_SERVER app patch ${{ inputs.service-identifier }}-${{ inputs.stage }} --patch='[{"op": "replace", "path": "/spec/info/0", "value": {"name": "Source Code Diff", "value": "'"https://github.com/${{ github.repository }}/compare/${OLD_TAG::8}...${GITHUB_SHA::8}"'"}}]'
+          ./argocd --grpc-web --auth-token $ARGOCD_TOKEN --server $ARGOCD_SERVER app set argocd/${{ inputs.service-identifier }}-${{ inputs.stage }} -p kotlin.revision=${GITHUB_SHA::8} -p kotlin.build=${{ github.run_number }} -p kotlin.image.tag=${GITHUB_SHA}
```
Comment (HotThoughts) [inline .github/workflows/deploy.yaml:622]: the way you patch source code diff link is brilliant 😍

Comment (raakatz) [reply .github/workflows/deploy.yaml:622]: Thanks! :D

**Review by HotThoughts — APPROVED** [review]

**Pull request merged**